src/nimble.nim

```diff
@@ -1994,6 +1995,36 @@ proc setup(options: Options) =
   setupNimbleConfig(options)
   setupVcsIgnoreFile()
 
+proc getAlteredPath(options: Options): string =
+  let pkgInfo = getPkgInfo(getCurrentDir(), options)
+  var pkgs = pkgInfo.processAllDependencies(options).toSeq.toOrderedSet
+  pkgs.incl(pkgInfo)
+
+  result = getEnv("PATH")
+  for pkg in pkgs:
+    let fullInfo = pkg.toFullInfo(options)
+    for bin, _ in fullInfo.bin:
+      let folder = fullInfo.getOutputDir(bin).parentDir.quoteShell
+      result = fmt "{folder}{separator}{result}"
+
+proc shellenv(options: var Options) =
+  setVerbosity(SilentPriority)
+  options.verbosity = SilentPriority
+  const prefix = when defined(windows): "set PATH=" else: "export PATH="
+  echo prefix & getAlteredPath(options)
+
+proc shell(options: Options) =
+  putEnv("PATH", getAlteredPath(options))
+
+  when defined windows:
+    var shell = getEnv("ComSpec")
+    if shell == "": shell = "powershell"
+  else:
+    var shell = getEnv("SHELL")
+    if shell == "": shell = "/bin/bash"
+
+  discard waitForExit startProcess(shell, options = {poParentStreams})
```

Comment:
On POSIX systems, using `execve` (or a similar function) would reduce the system memory usage here:

https://man7.org/linux/man-pages/man2/execve.2.html

Reply:
I was under the assumption that `poParentStreams` will end up using that. I will check that out.

Reply:
No, you are seeing the classic [fork/exec pattern](https://en.wikipedia.org/wiki/Fork%E2%80%93exec). 

No invocation of `startProcess` will be equivalent to `execve` because `startProcess` as an API returns the result of the launched process and the execution of the original program continues. The point of `execve` is to essentially terminate the execution of the current program by replacing its address space with the launched new program.